zng/resolver/encoder.go

```diff
@@ -50,7 +50,7 @@ func (e *Encoder) isEncoded(id int) bool {
 // the new type into the buffer provided.
 func (e *Encoder) Encode(dst []byte, external zng.Type) ([]byte, zng.Type, error) {
 	dst, typ, err := e.encodeType(dst, external)
-	if err != nil {
+	if err == nil {
```

Comment:
Would you please flip the logic around so this looks like the usual `if err != nil {return err}`?

Reply:
That would like like this which seems unnecessarily redundant:
```
if err != nil {
  return dst, typ, err
}
e.enter(...)
return dst, typ, err
```

Reply:
The error return can supply nils:

```
if err != nil {
  return nil, nil, err
}
```

Which is conventional, and will work for the only caller of Encode (in bzngio.Writer.Write).